### ISSUE TITLE
Urgent: Regular updates required

### ISSUE THREAD

**Issue by zb-markus** [klokantech/kartenportal-tickets#118]
We just noted that no maps catalogued after 19 August are contained in Geosearch. Reproduction:

NEBIS: search with words "raumentwicklung übersichtsplan" -> 72 records, catalogued between 16 and 28 August 2013
Swissbib: search with words "raumentwicklung übersichtsplan" -> 73 records
Geosearch: http://search.kartenportal.ch/#bbox=8.430939960938303,47.240761032351905,9.209021556615084,47.70664639790452&q=übersichtsplan publisher:raumentwicklung &date_from=2010&date_to=9999&scale_from=&scale_to=&libraries=&map=road -> 5 records

This is a major bug. Regular updates at least once per week are required.

Note:
Sample record:
http://www.swissbib.ch/TouchPoint/perma.do?q=0%3D%22220695520%22
where this does not exists:
http://search.kartenportal.ch/#id=220695520

**Comment by klokan** [comment]
The regular updates run every night - this is for example why #120 propagated the changes immediately.

There must be something special about these records - either there is a problem in MARC or on the side of SwissBib OAI-PMH export. We are investigating this and provide a report and fix.

Thank you for a nice bug report.

**Comment by zb-markus** [comment]
OK, we are awaiting your further investigation on this issue. 

On the other hand, we are most grateful that the importing script is running every night. This is clearly an improvement against the Lastenheft.

**Comment by klokan** [comment]
We have restarted downloading of the complete SwissBib (over 8GB of XML) again, it takes several hours - it has been finished once on Friday and once with a different logging system yesterday.

Our updating script runs every night. The script asks for changes since the last successful update - so normally for changes during last 24 hours - and downloads only the differences (a few KB to MB) and is finished usually in minutes. That's why we can run it every night.

Initially, we thought there were issues with SwissBib antedating the editing time - which would explain what has happened. I have made a draft for explaining the theory (https://gist.github.com/klokan/92f917573cc76f5a3629).

Today while further investigating we have found another bug caused by a typo in the OAI updating script on our side. We were wrongly setting the date of the "last successful update" in the updating script - so some of the dates with metadata updates were unfortunately skipped.

This happened as human mistake, because we had to re-index and completely download the whole OAI-PMH dump for almost every issue with metadata reported recently, and the same updating script has been modified there and back. The complete updating operation takes over 4 hours and downloads Gigabytes of data. After it is finished we were manually reverting it to the standard updating mode - and that's where the error happened.

In total we have spend over 7 hours on investigating this issue in two people.

Now the updating script should be correct.
The test record is available via Kartenportal.
The mentioned full text query seems to return in Kartenportal the correct amount of results as well.
